.NET Core - compile errors in Mono build

```
Compatibility/CustomAttributeExtensions.cs(32,37): error CS0030: Cannot convert type `System.Attribute[]' to `System.Collections.Generic.IEnumerable<T>'
Compatibility/CustomAttributeExtensions.cs(37,37): error CS0030: Cannot convert type `System.Attribute[]' to `System.Collections.Generic.IEnumerable<T>'
Core/Internal/AttributesUtil.cs(47,47): error CS0030: Cannot convert type `object[]' to `System.Collections.Generic.IEnumerable<T>'
Core/Internal/AttributesUtil.cs(49,46): error CS0030: Cannot convert type `object[]' to `System.Collections.Generic.IEnumerable<T>'
Core/Internal/AttributesUtil.cs(71,35): error CS0030: Cannot convert type `object[]' to `System.Collections.Generic.IEnumerable<T>'
Core/Internal/AttributesUtil.cs(73,34): error CS0030: Cannot convert type `object[]' to `System.Collections.Generic.IEnumerable<T>'
```

It seems that Mono compiler doesn't support array => IEnumerable<T> conversion.  I am working on a solution of `yield return` each of the array elements.

--- a/src/Castle.Core/Compatibility/CustomAttributeExtensions.cs
+++ b/src/Castle.Core/Compatibility/CustomAttributeExtensions.cs
@@ -29,12 +29,18 @@ namespace System.Reflection
 	{
 		public static IEnumerable<T> GetCustomAttributes<T>(this Assembly element) where T : Attribute
 		{
-			return (IEnumerable<T>)Attribute.GetCustomAttributes(element, typeof(T));
+			foreach (T a in Attribute.GetCustomAttributes(element, typeof(T)))
+			{
+				yield return a;
+			}
 		}
 
 		public static IEnumerable<T> GetCustomAttributes<T>(this MemberInfo element, bool inherit) where T : Attribute
 		{
-			return (IEnumerable<T>)Attribute.GetCustomAttributes(element, typeof(T), inherit);
+			foreach (T a in Attribute.GetCustomAttributes(element, typeof(T), inherit))
+			{
+				yield return a;
+			}
 		}
 
 		public static bool IsDefined(this MemberInfo element, Type attributeType)

--- a/src/Castle.Core/Core/Internal/AttributesUtil.cs
+++ b/src/Castle.Core/Core/Internal/AttributesUtil.cs
@@ -44,9 +44,18 @@ namespace Castle.Core.Internal
 		{
 			if (typeof(T) != typeof(object))
 			{
-				return (IEnumerable<T>)type.GetTypeInfo().GetCustomAttributes(typeof(T), false);
+				foreach (var a in type.GetTypeInfo().GetCustomAttributes(typeof(T), false))
+				{
+					yield return (T)a;
+				}
 			}
-			return (IEnumerable<T>)type.GetTypeInfo().GetCustomAttributes(false);
+			else
+			{
+				foreach (var a in type.GetTypeInfo().GetCustomAttributes(false))
+				{
+					yield return (T)a;
+				}
+			}
 		}
 
 		/// <summary>
@@ -68,9 +77,18 @@ namespace Castle.Core.Internal
 		{
 			if (typeof(T) != typeof(object))
 			{
-				return (IEnumerable<T>)member.GetCustomAttributes(typeof(T), false);
+				foreach (var a in member.GetCustomAttributes(typeof(T), false))
+				{
+					yield return (T)a;
+				}
 			}
-			return (IEnumerable<T>)member.GetCustomAttributes(false);
+			else
+			{
+				foreach (var a in member.GetCustomAttributes(false))
+				{
+					yield return (T)a;
+				}
+			}
 		}
 
 		/// <summary>

